### PR TITLE
Ridley Leisy - Classification 2

### PR DESCRIPTION
**Pull request by RidleyLeisy**

### CODE REVIEW

**Review by brennerhaverlock — CHANGES_REQUESTED** [review]
Hey Ridley great job importing glob and taking in the data

**Comment by axrd** [comment]
# Module 01 - Making Datasets, Avoiding Leakage

*Overall:* You're off to a great start with the Lending Club data! I can tell that having previous experience working locally is paying off. Please be on the lookout for students who aren't as familiar and try and help them as best you can!
*Feedback*: Get in the habit of being explicit about what you're doing and why. If you can explain your reasoning to a non-technical reader, you're quality of communication accounts for virtually every stakeholder you'll engage with. Leverage comments and text cells!

## Get the data 
- Glob FTW! Remember that there are some differences in data if you have an account vs not. 
- Really clever assigning your path to a variable to make your life easier 👍 

## Choose number of observations
- I see that you split the data but it would be great to add a text cell or some commenting to explain how/why you decided to do so (code etiquette). 

## Save your Data for reuse
- 👍 

## Choose Features
- Same as above. Document your choices more explicitly with text cells / comments.

**Comment by RidleyLeisy** [comment]
Sounds good!

**Comment by axrd** [comment]
# SC Review 

*Overall:* Great work this week. 
*Feedback*: Please keep your NB & Repos organized 😅 It's hard to figure out your understanding of the material otherwise. 

## Making Datasets, Avoiding Leakage
- Glob is a superpower. Remember that tomorrow in case you need to combine multiple csvs.  
- Key point of Data Hygiene in predictive modeling: leakage can easily lead you astray so proactively prevent it. Removing features is one really effective way to go about it. Consider Time-Series!

## Start with Baselines
- Not an explicit learning objective of this Sprint but so important that it's worth reiterating 👍 

## Class Imbalances
- When confronting classification problems, accuracy may not be a good measure because it doesn't factor True Positive rate. But the threshold for this is _contextual_. You want to bear in mind the *consequences* of your choices when factoring this in. Par for the course re: Predictive Modeling & Decision Making.
- If you determine that you can't afford false positives, use ROC AUC as a more appropriate metric for evaluating your model. By plotting the true positive rate vs the false positive rate at different thresholds, you can better understand your model's predictive power. 

## Making & Explaining Decisions
- The whole point of Data Science (if you can reduce it down to just one) is to draw actionable insights. That means making decisions. In order to better calibrate that decision-making, we leverage Expected Values, and Shapley Value Plots. Review the lecture notebook for links to dive deeper into both of these topics if you wish!

## Gradient Descent
- How can you programaticaly minimize a value you care about? Gradient Descent refers to a method that iteratively discovers minima. The trick is understanding if they are local or global, for one. For another, computation is not free. The time and effort it takes your model to minimize it's errors is a consideration you must make. Calibrate the learning_rate as one knob/dial.

**Comment by RidleyLeisy** [comment]
Thanks!
